app/src/main/AndroidManifest.xml

```diff
@@ -2,6 +2,9 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools">
 
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
```

Comment:
yes you are right if we follow the principle of least privilege it would just be necessary to have the fine location. I will check that

Reply:
I checked and I have the impression that Android Studio is not happy if I remove ACCESS_COARSE_LOCATION. So I'm going to leave both permissions. Either way, from a security perspective this is not really a problem because if fine location is allowed, coarse location does not add any additional exposure